Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1111 @@
+      · rintro ⟨y, rfl⟩
+        rw [Set.mem_preimage]
+        refine' ⟨⟨algebraMap R S y, spectrum.algebraMap_mem R S y.prop⟩, _⟩
+        simp only [ContinuousMap.coe_mk, ContinuousMap.comp_apply, StarAlgHom.ofId_apply]
+        congr
+        exact Subtype.ext (h.left_inv y)
+    hom_closedEmbedding := by
+      apply ClosedEmbedding.comp (cfc₁_closedEmbedding S a)
+      simp only [AlgHom.coe_toRingHom, StarAlgHom.coe_toAlgHom, StarAlgHom.comp_apply,
+        ContinuousMap.compStarAlgHom'_apply, ContinuousMap.compStarAlgHom_apply]
+      have : CompactSpace (spectrum R a) := SpectrumRestricts.compactSpace f
+      refine Isometry.closedEmbedding ?_
+      simp only [isometry_iff_dist_eq]
+      intro g₁ g₂
+      refine' le_antisymm _ _
+      · rw [ContinuousMap.dist_le dist_nonneg]
+        intro x
+        simp only [ContinuousMap.coe_mk, ContinuousMap.comp_apply, StarAlgHom.ofId_apply]
+        rw [h_isom.dist_eq]
+        exact ContinuousMap.dist_apply_le_dist _
+      · rw [ContinuousMap.dist_le dist_nonneg]
+        intro x
+        obtain ⟨y, y_mem, hy⟩ := (h.image.symm ▸ x.prop : (x : R) ∈ f '' spectrum S a)
+        lift y to spectrum S a using y_mem
+        refine le_of_eq_of_le ?_ <| ContinuousMap.dist_apply_le_dist y
+        simp only [ContinuousMap.coe_mk, ContinuousMap.comp_apply, StarAlgHom.ofId_apply]
+        rw [h_isom.dist_eq]
+        congr <;> exact Subtype.ext hy.symm }
+
+/-- If the spectrum of `a` restricts from `S` to `R`, then so does `cfc₁ g` for any
+`g : C(spectrum R a, R)`. You should use this lemma manually to prove the spectrum restriction
+result for continuous functional calculi whenever you use one of the definitions above to create an
+instance.
+
+Tou can use this to prove that, for exmaple, the spectrum (in `ℂ`) of the image of a positive
+operator is nonnegative. -/
+theorem SpectrumRestricts.cfc_spectrumRestricts {R : Type u} {S : Type v} {A : Type w}
+    [CommSemiring R] [StarRing R] [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R]
+    [CommSemiring S] [StarRing S] [MetricSpace S] [TopologicalSemiring S] [ContinuousStar S]
+    [Ring A] [StarRing A] [TopologicalSpace A] [Algebra R S] [Algebra R A] [Algebra S A]
+    [IsScalarTower R S A] [StarModule R S] [ContinuousSMul R S] {a : A} [CFCClass S a]
+    (f : C(S, R)) (h : SpectrumRestricts a f) (g : C(spectrum R a, R)) :
+    SpectrumRestricts (@cfc₁ _ _ _ _ _ _ _ _ _ _ _ _ (h.cfcCore f) g) f :=
+  { rightInvOn := by
+      intro s hs
+      erw [h.starAlgHom_apply, cfc₁_map_spectrum] at hs
+      obtain ⟨x, hx⟩ := hs
+      simp only [ContinuousMap.coe_mk, ContinuousMap.comp_apply, StarAlgHom.ofId_apply] at hx
+      nth_rw 1 [← hx]
+      rwa [h.left_inv]
+    left_inv := h.left_inv }
+
+end Universal
+
+section ComplexToReal
+
+noncomputable instance CfcCore.complexToReal {A : Type*} [Ring A] [StarRing A] [TopologicalSpace A]
+    [Algebra ℂ A] {a : A} [CFCCoreClass ℂ a] [h : SpectrumRestricts a ContinuousMap.complexRe] :
```

Comment:
```suggestion
noncomputable instance CfcCore.complexToReal {A : Type*} [Ring A] [StarRing A] [TopologicalSpace A]
    [Algebra ℂ A] {a : A} [CFCCoreClass ℂ a] [h : SpectrumRestricts a ContinuousMap.complexRe] :
```
Shouldn't it be `CFC` instead of `Cfc` here and below?